packages/manager/apps/telecom/src/app/telecom/telephony/line/phone/accessories/choice/telephony-line-phone-accessories-choice.controller.js

```diff
@@ -0,0 +1,165 @@
+import chunk from 'lodash/chunk';
+import find from 'lodash/find';
+import sortBy from 'lodash/sortBy';
+
+import initializeBrandList from './telephony-line-phone-accessories-choice.service';
+
+export default class TelecomTelephonyLinePhoneAccessoriesChoiceCtrl {
+  /* @ngInject */
+  constructor($q, $translate, TucTelephonyAccessoriesOrderProcess) {
+    this.$q = $q;
+    this.$translate = $translate;
+    this.TucTelephonyAccessoriesOrderProcess = TucTelephonyAccessoriesOrderProcess;
+  }
+
+  $onInit() {
+    this.process = null;
+    this.orderTotal = null;
+
+    this.loading = {
+      init: false,
+    };
+
+    this.error = {
+      loading: null,
+    };
+
+    this.accessoryToOrder = null;
+    this.selectedAccessory = null;
+
+    this.spinnerExtremities = {
+      min: 0,
+      max: 5,
+    };
+
+    this.init();
+  }
+
+  init() {
+    this.loading.init = true;
+
+    return this.TucTelephonyAccessoriesOrderProcess.getAvailableAccessories()
+      .then(
+        (orderProcess) => {
+          this.process = orderProcess;
+          this.brandList = initializeBrandList(this.process.accessoriesList);
+          this.accessoriesDisplayed = this.process.accessoriesList;
+          this.chunkedList = chunk(this.process.accessoriesList, 4);
+          this.orderTotal = this.TucTelephonyAccessoriesOrderProcess.getPriceStruct(
+            0,
+          );
+        },
+        (error) => {
+          this.error.loading = error;
+          return this.$q.reject(error);
+        },
+      )
+      .finally(() => {
+        this.loading.init = false;
+      });
+  }
+
+  hasAtLeastOneAccessory() {
+    return !!find(
+      this.process.accessoriesList,
+      (accessory) => accessory.quantity > 0,
+    );
+  }
+
+  updateOrderTotal(quantity, accessoryName) {
+    let total = 0;
+    angular.forEach(this.process.accessoriesList, (accessory) => {
+      if (accessory.name === accessoryName) {
+        total += accessory.price.value * quantity;
+      } else {
+        total += accessory.price.value * accessory.quantity;
+      }
+
+      if (quantity === 5 || accessory.quantity === 5) {
+        this.maxQuantity = true;
+      } else {
+        this.maxQuantity = false;
+      }
+    });
+
+    this.orderTotal = this.TucTelephonyAccessoriesOrderProcess.getPriceStruct(
+      total,
+    );
+
+    return this.orderTotal;
+  }
+
+  filterByBrand(brand) {
+    const listFiltered = [];
+    if ('all'.includes(brand.toLowerCase())) {
```

Comment:
What are the possible values for brand ? Could be a strict equality check ? 🤔

Reply:
I don't have a fix list of brands, I just build the list from the result of the API. And I check a strict equality to filter the list of phones or accessories by brand to display.